src/wrappers/amqplib.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @fileoverview Handlers for amqplib instrumentation
+ */
+
+const {
+    tracer,
+    moduleUtils,
+    eventInterface,
+    utils,
+} = require('epsagon');
+const traceContext = require('../trace_context.js');
+const { EPSAGON_HEADER } = require('../http.js');
+
+/**
+ * acts as a middleware for `consumer.run()`
+ * @param {object} message the messages param to send
+ * @param {Function} callback the callback function
+ * @param {Channel} channel the Channel object of amqplib
+ * @param {function} originalHandler original consumer function
+ */
+function amqplibSubscriberMiddleware(message, callback, channel) {
+    let originalHandlerSyncErr;
+    try {
+        // Initialize tracer and runner.
+        tracer.restart();
+        const { slsEvent: amqpEvent, startTime: amqpStartTime } =
+        eventInterface.initializeEvent(
+            'rabbitmq',
+            message.fields.routingKey,
+            'consume',
+            'trigger'
+        );
+
+        const metadata = {
+            exchange: message.fields.exchange,
+            redelivered: message.fields.redelivered,
+            host: channel.connection.stream._host, // eslint-disable-line no-underscore-dangle
+            consumer_tag: message.fields.consumerTag,
+        };
+        if (message.properties.headers[EPSAGON_HEADER]) {
+            metadata[EPSAGON_HEADER] = message.properties.headers[EPSAGON_HEADER].toString();
+        }
+
+        tracer.addEvent(amqpEvent);
+        eventInterface.finalizeEvent(amqpEvent, amqpStartTime, null, metadata, {
+            headers: message.properties.headers,
+            message: message.content.toString(),
+        });
+
+        const { label, setError } = tracer;
+        // eslint-disable-next-line no-param-reassign
+        message.epsagon = {
+            label,
+            setError,
+        };
+        const { slsEvent: nodeEvent, startTime: nodeStartTime } = eventInterface.initializeEvent(
+            'node_function', 'message_handler', 'execute', 'runner'
+        );
+        let runnerResult;
+        try {
+            runnerResult = callback(message);
+        } catch (err) {
+            originalHandlerSyncErr = err;
+        }
+
+        if (callback.name) {
```

Comment:
ahh i see you do it here, anyway add the `<queue name>-consumer` (or something similar) option. Maybe also consult with @ronyis

Reply:
👍